eppo_core/src/attributes.rs

```diff
@@ -67,19 +108,19 @@ mod pyo3_impl {
     impl<'py> FromPyObject<'py> for AttributeValue {
         fn extract_bound(value: &Bound<'py, PyAny>) -> PyResult<AttributeValue> {
             if let Ok(s) = value.downcast::<PyString>() {
-                return Ok(AttributeValue::String(s.extract()?));
+                return Ok(AttributeValue::String(s.to_cow()?.into()));
```

Comment:
Trying to understand the inner workings of Rust here, hope you are willing to entertain me. According to ChatGPT:

```
If the string is always going to be used as owned data, you could bypass the Cow<str> entirely and directly convert it into a String. However, the current approach is more flexible and performance-conscious, as it handles both borrowed and owned cases efficiently.
```

Do we need the flexibility to handle both? I figured we probably want to own it here regardless? I might be misunderstanding

Reply:
The answer is correct for `Cow<str> -> String` conversion, however we're going from `Cow<str> -> FastStr` which may skip memory allocation for strings of up to 24 bytes (when converting from `Cow::Borrowed`).

Another important reason why we're using `.to_cow()` here is because it's the most performant way to get a string from Python with pyo3. Python 3.10 API introduced new functions to get the string reference directly (which would return `Cow::Borrowed` here) and for Python versions before that, it needs to copy the string ([source](https://github.com/PyO3/pyo3/blob/95949944554354cdbf3275daa03818a6f5b65c33/src/types/string.rs#L333-L348)).